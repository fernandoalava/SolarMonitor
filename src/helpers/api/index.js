import config from "../../config";
import axios from "axios";

export const cloudAPIUrl = `${config.protocol}://${
  config.planetOSDomain
}/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
  config.apiKey
}&count=50&lat=49.6&lon=-50.5&var=av_ttl_cld`;

export const solarAPIUrl = `${config.protocol}://${
  config.planetOSDomain
}/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
  config.apiKey
}&count=50&lat=49.6&lon=-50.5&var=av_swsfcdown`;

export const fetchCloudData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(cloudAPIUrl);
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};

export const fetchSolarData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(solarAPIUrl);
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};
