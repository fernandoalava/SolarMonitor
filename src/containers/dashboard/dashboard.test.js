import { shallow } from "enzyme";
import * as React from "react";
import { DashBoardScreen } from ".";
import { mockPanelsData } from "../../helpers";

jest.mock("../../helpers");
const { getPanelsInfo } = require("../../helpers");
getPanelsInfo.mockImplementation(() => mockPanelsData);

jest.useFakeTimers();

describe("Dashboard", () => {
  it("Renders correctly", () => {
    const wrapper = shallow(<DashBoardScreen />);
    expect(wrapper).toMatchSnapshot();
  });
});
