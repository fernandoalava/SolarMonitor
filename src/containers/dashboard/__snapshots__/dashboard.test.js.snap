// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Renders correctly 1`] = `
<styled.section>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
    styles={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="flex-start"
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <TotalEnergyCard
            data={Array []}
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <SolarActivity
            activity={Array []}
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <CloudCoverage
            activity={Array []}
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="flex-start"
        spacing={2}
      >
        <SolarPanelsTable
          data={Array []}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</styled.section>
`;
